### PR TITLE
Update boto_dynamodb module and state

### PR DESCRIPTION
**Pull request by zackhsi**
- support updating adjusting provisioned throughput of an existing table
- manage periodic backups via boto_datapipeline
- local secondary index bugfix

### CODE REVIEW

File: salt/modules/boto_dynamodb.py [hunk 1]
```diff
@@ -74,7 +74,7 @@ def __virtual__():
     Only load if boto libraries exist.
     '''
     if not HAS_BOTO:
-        return (False, 'The module boto_dynamodb could not be loaded: boto libraries not found')
```
Comment (ryan-lane) [inline salt/modules/boto_dynamodb.py]: I think this was a change in develop that we want to keep.

Comment (zackhsi) [inline salt/modules/boto_dynamodb.py]: Done

**Comment by cachedout** [comment]
Hi @zackhsi 
Thanks for this!

There is a failing test here: https://jenkins.saltstack.com/job/salt-pr-rs-cent7-n/11336/testReport/junit/unit.states.boto_dynamodb_test/BotoDynamodbTestCase/test_present/

Could you please take a look before we merge this? Thanks.

**Comment by zackhsi** [comment]
@cachedout updated!

**Comment by cachedout** [comment]
@zackhsi Great! Thank you.

**Pull request merged**